users/services/utils.py

```diff
@@ -25,7 +26,7 @@ def validate_password(password) -> dict:
     return validation_result
 
 
-def validate_email(email) -> bool:
+def validate_email(email) -> Optional[Match[str]]:
```

Comment:
If I may know, why are we required to change the return type from boolean to Optional[Match[str]]?

Reply:
Hello, Alex. This is because I decided to change the implementation by utilizing the regex library. Here, I return re.fullmatch which will contain a re.Match object when the pattern is found and None when the pattern is not found. Due to the nature of the return value, that may be None or a re.Match object, the Optional type is used to better describe the function.